src/app/(routes)/page.tsx

```diff
@@ -3,19 +3,38 @@
 import { Card } from "@/components/ui/card";
 import KycApproval from "@/components/ui/kycApproval";
 import { ConnectButton } from "@rainbow-me/rainbowkit";
-import { useEffect } from "react";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
 import { useSignTypedData, useAccount } from "wagmi";
 
 export default function Home() {
-  const { address, isConnecting, isDisconnected, isConnected } = useAccount();
-  console.log(address);
+  const { isConnected } = useAccount();
+  const [kycParams, setKycParams] = useState({ clientId: "", repoName: "", repoOwner: "" });
+  const router = useRouter();
+  const { query } = router;
+
+  useEffect(() => {
+    if (query.clientId && query.repoName && query.repoOwner) {
+      setKycParams({
+        clientId: Array.isArray(query.clientId) ? query.clientId[0] : query.clientId,
+        repoName: Array.isArray(query.repoName) ? query.repoName[0] : query.repoName,
+        repoOwner: Array.isArray(query.repoOwner) ? query.repoOwner[0] : query.repoOwner
```

Comment:
I'm not a FE expert, so treat this as legit question, not request for immediate change - why `useEffect` for this? I'd just do it as part of rendering, as this doesn't seem compute-heavy.
Even if it's compute-heavy, wouldn't `useMemo` be a better fit here?

Reply:
totally agree